### ISSUE TITLE
Ctrl-tab to switch files on Windows quite often causes Caret to hang

### ISSUE THREAD

**Issue by dracan** [careteditor/caret#539]
As per the title really. When I press `ctrl-tab`, I get the popup with the list of files, but then it quite often just hangs at this point.

I know this isn't just an issue on my machine, as chatting with an ex-colleague who'd given Caret a go, he mentioned this too.

**Comment by astoilkov** [comment]
We have other users reporting the same problem. We can't replicate the problem. We will release Caret 3.5 early next week which we think has a big chance of fixing such a problem.

However, if you see any pattern of when this happens this would be of great help. Maybe some specific file?

**Comment by dracan** [comment]
Not sure to be honest. It's done it most time I've tried to use ctrl-tab, so I've kind-of avoided using it. Typically, I've just tried it now, and it's not doing it! I think this is one of the first times I've seen it work as expected. Not sure what's different!

**Comment by thomas-sittler** [comment]
Yes, it crashes most times for me when i use ctrl-tab.

**Comment by dracan** [comment]
Is there anything further on this? It's not just ctrl-tab - it's also selecting an item from the file list. Crashes so often, I'd given up on Caret and gone back to VSCode for Markdown editing. I've just come back to it now to see if it's improved, but there's not even been an update to Caret since - so I'm assuming you're now just focusing on v4?

It's a real shame, as Caret is such a good looking Markdown editor, and has such a _really_ nice user interface. It just feels _really_ unstable. As if all the attention went into the user interface, and not stability. I really want to come back to Caret, and just hope v4 is more stable. Especially after paying for v3 and not really being able to use it without constantly swearing at it.

**Comment by dracan** [comment]
Actually, turns out it hadn't crashed - it just took about 2-3 minutes to load the file! Whilst I was typing the above message, it came back to life and loaded the file I had tried to load. So it must have been quite a few minutes later.

**Comment by astoilkov** [comment]
Hi @dracan,

Caret 4.0 has a big change of fixing this bug (I use the word "chance" because we actually never replicated this bug). We are working full time every day to bring Caret 4.0 to a Beta release. We are really close but we have a few more days until it's finished.

P.S. I want to say we are concentrating on every bug reported and want to make that app really stable. Our goal isn't making just the next Markdown editor. We want to make the best writing experience in the world but this takes time. We are currently concentrated on finding the best workflow for working with local files - Caret 4.0 is a really huge improvement on that. After we manage to achieve this we will concentrate all our efforts on refactoring and stability.

I hope you could understand our position.

**Comment by dracan** [comment]
Yep, completely understand. I'm a dev myself, so know how much of a PITA these bugs you can't repro can be. Really looking forward to seeing what you've done with Caret 4!

**Comment by astoilkov** [comment]
We released [Caret 4.0 Beta](https://caret.io/4beta/).

Is anyone willing to confirm if the Beta has fixed the issue?

Thanks in advance!

**Comment by thomas-sittler** [comment]
In the meantime I kinda stopped using Caret, because of too many such issues (switched to Sublime Text). But I'll try the Beta now.

**Comment by thomas-sittler** [comment]
Double clicking on files to open now seems to work without crashing (just a few tries). It becomes unresponsive (as in caret 3) when I try so switch between files from the sidebar. Definitely feels snappier when it's not crashing.

**Comment by dracan** [comment]
It's not crashing for me when ctrl tabbing any more in beta 4.

**Comment by thomas-sittler** [comment]
It is for me. I should say it hangs up (tied to load the file forever and becomes unresponsive) rather than it crashes.

**Comment by astoilkov** [comment]
@thomas-sittler I didn't understand your last comment. Sorry. Do you mean you are experiencing the same issue in the Beta?

**Comment by thomas-sittler** [comment]
Yes. As you can see in this video, with a few files open, after a bit of ctrl-tabbing, it starts to take several seconds to open a file, until it grinds to a halt completely. https://drive.google.com/open?id=19MIg6QG4X4S0uvFdh7AX6ZJmYxUMV8ff

**Comment by astoilkov** [comment]
@thomas-sittler Thanks for the video! Is it possible you send us the files from the video? This could help us a lot in replicating the problem.

If you decide you can share those documents you can send them to astoilkov@caret.io.

Thanks in advance.

**Comment by thomas-sittler** [comment]
You can find them here for example.
https://github.com/thomas-sittler/thomas-sittler.github.io/tree/master/_posts

On 12 Jan 2018 at 16:46, <Antonio Stoilkov <notifications@github.com>>
wrote:

@thomas-sittler <https://github.com/thomas-sittler> Thanks for the video!
Is it possible you send us the files from the video? This could help us a
lot in replicating the problem.

If you decide you can share those documents you can send them to
astoilkov@caret.io.

Thanks in advance.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/careteditor/caret/issues/539#issuecomment-357287627>,
or mute the thread

**Comment by astoilkov** [comment]
@thomas-sittler We are happy to say we found and fixed the problem you showed in the video. The next Beta release will include the fix - I will write in this thread when this happens.
